docs/SETUP.md

```diff
@@ -0,0 +1,236 @@
+1. `cp .env_sample .env`
+
+## On Windows
+
+*TODO describe how to install on Windows*
+
+## Editing project variables
+
+Open `.env` in your preferred editor, you'll fill this in with your storage keys, SSL information, etc.
+
+
+```ini
+# ----------------------------------------------------------------------------
+# Storage
+# ----------------------------------------------------------------------------
+
+# Uncomment to use AWS
+DEFAULT_FILE_STORAGE=storages.backends.s3boto.S3BotoStorage
+AWS_ACCESS_KEY_ID=
+AWS_SECRET_ACCESS_KEY=
+AWS_STORAGE_BUCKET_NAME=public
+AWS_STORAGE_PRIVATE_BUCKET_NAME=private
+AWS_S3_CALLING_FORMAT=boto.s3.connection.OrdinaryCallingFormat
+AWS_S3_HOST=s3-us-west-2.amazonaws.com
+AWS_QUERYSTRING_AUTH=False
+
+# Uncomment to use Azure
+#DEFAULT_FILE_STORAGE=codalab.azure_storage.AzureStorage
+AZURE_ACCOUNT_NAME=
+AZURE_ACCOUNT_KEY=
+AZURE_CONTAINER=public
+# Only set these if bundle storage key is different from normal account keys
+# BUNDLE_AZURE_ACCOUNT_NAME=
+# BUNDLE_AZURE_ACCOUNT_KEY=
+BUNDLE_AZURE_CONTAINER=bundles
+
+...
+```
+
+# 2. Storage 
+
+Codalab gives you the option of using AWS or Azure as your storage of choice. Depending on vendor you use, **you must comment out the one you are not using in the `.env` file.**
+
+## AWS S3
+
+Sign in or create an AWS account [here](https://aws.amazon.com/s3/) and then create a private and public bucket.
+
+
+### Creating buckets
+
+1. You don't have to do this if you've already setup Azure Blob Storage!
+1. Sign into the AWS Management Console and open the Amazon S3 console at [here](https://console.aws.amazon.com/s3.)
+1. Create 2 buckets, one named `"whatever-public"` and another `"whatever-private"`
+1. Click **Create Bucket**
+1. In the **Create a Bucket** dialog box, in the **Bucket Name** box, enter a bucket name
+1. In the **Region** box, select US West 2.
+1. Click **Create**
+1. Put these bucket names in your `.env` under `AWS_STORAGE_BUCKET_NAME` and `AWS_STORAGE_PRIVATE_BUCKET_NAME`
+1. Make sure the `DEFAULT_FILE_STORAGE` `.env` option is set to `storages.backends.s3boto.S3BotoStorage`
+
+### Setting CORS
+
+In both buckets set CORS as follows:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<CORSConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+    <CORSRule>
+        <AllowedOrigin>*</AllowedOrigin>
+        <AllowedMethod>PUT</AllowedMethod>
+        <AllowedMethod>POST</AllowedMethod>
+        <AllowedMethod>GET</AllowedMethod>
+        <MaxAgeSeconds>3000</MaxAgeSeconds>
+        <AllowedHeader>*</AllowedHeader>
+    </CORSRule>
+</CORSConfiguration>
+```
+
+![image](https://cloud.githubusercontent.com/assets/2185159/23929355/b0d1ce94-08e2-11e7-90af-02ee57d0fcf0.png)
+
+
+### Setting Bucket Policies
+
+If you don't already have a user (via Identity and Access Management -- IAM), you'll need to create that [here]() and put those key/secret values into your `.env` under `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
```

Comment:
![add user](https://cloud.githubusercontent.com/assets/13974084/24004744/809aaadc-0a3d-11e7-9c2a-df0a0d9c3813.png)
![programmatic access](https://cloud.githubusercontent.com/assets/13974084/24004748/866b4a3e-0a3d-11e7-8419-abb9841b5e8a.png)
![scary permissions box](https://cloud.githubusercontent.com/assets/13974084/24004793/aa1fbc58-0a3d-11e7-80ca-5f8ba9d9f668.png)
![access key and id](https://cloud.githubusercontent.com/assets/13974084/24004836/c6776f5e-0a3d-11e7-88bc-c49f25345907.png)